DESIGN.md

```diff
@@ -0,0 +1,60 @@
+# Bulk Data Extract (BDE) Processor design
+
+## User requirements
+
+-  The _user interface_ is a read-only PostgreSQL database in AWS RDS, with
+   Azure authentication
+-  _Accessible_ to all
+   [current BDE users](https://github.com/linz/ds-infra-team/wiki/BDE-users)
+-  Process BDE files
+   _[every day](https://toitutewhenua.atlassian.net/wiki/spaces/GEOD/pages/86418416/BDE+Processor+Product)
+   except Sunday_
+-  Optional _post-processing_ pulled in from separate internal repository
+-  All changes for a single day are imported within a _single transaction_
+-  _Immediate notification_ of completed import (like the legacy BDE Processor
+   SNS topic)
+
+## Technical requirements
+
+-  Thorough testing of every part of the process before merging any pull
+   requests
+-  Separate, managed accounts for infrastructure testing and production
+-  Trivial deployment
+
+## Open questions
+
+-  Can we give read access to the BDE data to the entire organisation? The
+   public?
+
+## Non-requirements
+
+-  Don't keep BDE files around. They should ideally be kept only in memory or at
+   most in temporary storage which is cleaned up after each run. If we
+   absolutely need to keep them around that should be done as a separate system.
+   If it's convenient, the BDE Processor could read from the data in that system
+   rather than doing its own data copying.
+-  No table versioning. It creates a _lot_ of extra complexity, and we don't
```

Comment:
Two things here.

1. We need to check what issues this will cause for LINZ users, i.e. positioning team etc.
2. In the case of needing to get a version of the data at some specific point in time. How would we go about that? Could we just reprocess some old files from the archive for example?

Reply:
We'd need to check with the relevant teams how they use versioned data, and why. Reprocessing would be an option, but I'd rather not store many copies of the same data if avoidable.